grafana/grafana
# PublicDashboards: Modal warns when using unsupported datasources

fixes https://github.com/grafana/grafana-enterprise/issues/4452

**What this does**
When you're using a datasource that's not supported by pubdash (alerting and backend are not both true), you'll get a warning on the modal listing which datasources are not supported and a link to the docs to see which datasources are supported (coming in v9.3). You can still save the pubdash when the warning is there.

<img width="687" alt="Screenshot 2022-11-16 at 2 09 34 PM" src="https://user-images.githubusercontent.com/20195316/202520256-a54a3ceb-0445-4e4b-be14-a9990fdab7dd.png">

**How to test**
Easiest way is to go to the `SupportedPubdashDatasources` file and comment out the datasources you're using in your dashboard. Then open up the modal and you should see the warning.

**Design Notes**
I opted to generate the map of supported datasources on the frontend for this use case. It didn't make any sense to have to make another request to the backend to do it. Its pretty soft validation anyways, so doing it on the frontend is fine imo. If someone wants to ignore the warning, its totally fine.
